### PR TITLE
Fix some tests and dependencies

### PR DESCRIPTION
**Pull request by freiksenet**

### CODE REVIEW

File: packages/gatsby/src/schema/directives/__tests__/dateformat.js [hunk 1]
```diff
@@ -99,8 +99,8 @@ describe(`@dateformat directive`, () => {
   })
 
   it(`uses default directive args`, async () => {
-    const date = new Date(Date.UTC(2019, 0, 1))
-    Date.now = jest.fn().mockReturnValue(new Date(Date.UTC(2019, 0, 3)))
+    const date = new Date(2019, 0, 1, 10)
```
Comment (freiksenet) [inline packages/gatsby/src/schema/directives/__tests__/dateformat.js:102]: With UTC results are different based on machine timezone.

**Comment by freiksenet** [comment]
Added example project.

We should keep the original arguments for fields for now, everything is moved under filter, file used to have relativePath, we shouldn't break small things like that.

Maybe we should have @fileFromRelativePath decorator for explicit `File`s in type defis.

File: packages/gatsby/src/schema/schema/index.js [hunk 2]
```diff
@@ -13,13 +13,7 @@ const addResolvers = require(`./add-resolvers`)
 const addThirdPartySchemas = require(`./add-third-party-schemas`)
 
 const addTypeDefs = typeDefs => {
-  const types = schemaComposer.addTypeDefs(typeDefs)
-  types.forEach(type => {
-    const tc = schemaComposer.getTC(type.name)
```
Comment (stefanprobst) [inline packages/gatsby/src/schema/schema/index.js:18]: IIRC I kept this here (before `addInferredTypes`) so we can bail early there and not try to infer fields from `internal` etc.

**Comment by stefanprobst** [comment]
Oh how exciting! My first PR ever! Thanks! One comment above.

**Comment by stefanprobst** [comment]
I am working on bringing this branch into shape and restore current behavior, should be finished on the weekend at the latest.

File: packages/gatsby/src/schema/schema/index.js (hunk 2 above)
Comment (freiksenet) [reply packages/gatsby/src/schema/schema/index.js:18]: This forces schema composer too early, eg it fails if you refer to File type inside your typedefs when File type isn't inferred yet.

**Comment by stefanprobst** [comment]
Thanks @freiksenet!

File: packages/gatsby/src/schema/directives/__tests__/dateformat.js (hunk 1 above)
Comment (stefanprobst) [reply packages/gatsby/src/schema/directives/__tests__/dateformat.js:102]: The default value for the `difference` field needed changing to local time as well. Should be good now.

**Pull request merged**